person.rb

```diff
@@ -23,3 +30,33 @@ def of_age?
     @age > 18
   end
 end
+
+class Decorator < Person
+  def initialize(person)
+    super
+    @person = person
+  end
+
+  def correct_name
+    @person.name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @person.name.upcase
+  end
+end
```

Comment:
- [ ] Upon addressing what I pointed out as regards the Decoration class, you will need to alter this method too as you will work with @nameable and not @person. Something more like `@nameable.correct_name.capitalize`